### PR TITLE
Improved Server Addresses Collection Debugging

### PR DESCRIPTION
**Pull request by MythoclastBM**
Added new class to use as DebuggerTypeProxy in ServerAddressesCollection for Kestrel. Also modified the solution file for Kestrel as attempting to build any Sample Projects would throw a FileNotFoundException for Mvc.ApiExplorer, though this can be removed if incorrect or undesired. 

This corresponds to helpwanted issue 54030: 
https://github.com/dotnet/aspnetcore/issues/54030

### CODE REVIEW

**Comment by MythoclastBM** [comment]
@dotnet-policy-service agree

**Review by JamesNK — APPROVED** [review]

File: src/Servers/Kestrel/Core/src/Internal/ServerAddressesCollection.cs [hunk 1]
```diff
@@ -169,6 +171,12 @@ private void ThrowIfReadonly()
         }
     }
 
+    private sealed class PublicServerAddressesCollectionDebugView(PublicServerAddressesCollection publicCollection)
+    {
+        private readonly PublicServerAddressesCollection _collection = publicCollection;
+        [DebuggerBrowsable(DebuggerBrowsableState.RootHidden)]
+        public string[] Items => _collection.ToArray();
+    }
```
Comment (joegoldman2) [inline src/Servers/Kestrel/Core/src/Internal/ServerAddressesCollection.cs:179]: nit:
```suggestion
    }

```

File: src/Servers/Kestrel/Kestrel.slnf [hunk 2]
```diff
@@ -26,8 +26,13 @@
       "src\\Middleware\\Diagnostics\\src\\Microsoft.AspNetCore.Diagnostics.csproj",
       "src\\Middleware\\HostFiltering\\src\\Microsoft.AspNetCore.HostFiltering.csproj",
       "src\\Middleware\\HttpOverrides\\src\\Microsoft.AspNetCore.HttpOverrides.csproj",
+      "src\\Middleware\\OutputCaching\\src\\Microsoft.AspNetCore.OutputCaching.csproj",
+      "src\\Middleware\\ResponseCaching.Abstractions\\src\\Microsoft.AspNetCore.ResponseCaching.Abstractions.csproj",
       "src\\Middleware\\StaticFiles\\src\\Microsoft.AspNetCore.StaticFiles.csproj",
       "src\\Middleware\\WebSockets\\src\\Microsoft.AspNetCore.WebSockets.csproj",
+      "src\\Mvc\\Mvc.Abstractions\\src\\Microsoft.AspNetCore.Mvc.Abstractions.csproj",
```
[the remark this answers is not in the document]
Comment (MythoclastBM) [reply src/Servers/Kestrel/Kestrel.slnf]: Removed

**Comment by JamesNK** [comment]
Thanks for contributing this improvement!

**Pull request merged**